src/main/java/com/minecolonies/coremod/entity/pathfinding/pathjobs/AbstractPathJob.java

```diff
@@ -709,12 +735,7 @@ private void handleDebugDraw()
     {
         if (debugDrawEnabled)
         {
-            synchronized (debugNodeMonitor)
-            {
-                lastDebugNodesNotVisited = debugNodesNotVisited;
-                lastDebugNodesVisited = debugNodesVisited;
-                lastDebugNodesPath = debugNodesPath;
-            }
+            synchToClient(entity.get());
```

Comment:
this can ran into concurrency issues due to beeing called on the pathfinding thread and the entity tracking/player map could've been updated in the meanwhile, maybe move it into the navigator getting the future result and then make it accessible through the job? future -> pathresult -> job -> debugNodes -> send

Reply:
I wonder if a concurrentMap for the tracking is neough or if the network access off thread can cause issues